src/main/java/org/apache/sling/distribution/impl/DefaultDistributor.java

```diff
@@ -31,22 +28,22 @@
 import org.apache.sling.distribution.component.impl.DistributionComponentProvider;
 import org.apache.sling.distribution.common.DistributionException;
 import org.jetbrains.annotations.NotNull;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 /**
  * Default implementation of Distributor interface that dispatches the request to available agents.
  */
-@Component
-@Service(Distributor.class)
+@Component(service=Distributor.class)
```

Comment:
IIRC you don't need `service=...` here as it's inferred.

Reply:
fixed